src/main/flight/gps_rescue.c

```diff
@@ -233,9 +221,7 @@ static void rescueAttainPosition(void)
     /**
         Altitude (throttle) controller
     */
-
-    const float verticalVelocity = getAltitudeDerivative() * rescueState.intent.throttleDMultiplier;
-    altitudeControl(rescueState.intent.targetAltitudeCm, taskIntervalSeconds, verticalVelocity, rescueState.intent.targetAltitudeStepCm);
+    altitudeControl(rescueState.intent.targetAltitudeCm, taskIntervalSeconds, rescueState.intent.targetAltitudeStepCm);
```

Comment:
In a sense that's exactly what this does.  It sends the target altitude.

We could send a target xyz location, but for now it's simpler for my small brain, in the first iteration, that some semblance of the 'old way' remains, and that kept altitude separate from position.

I appreciate that it would be nice to just send a gps_location_t value, but first I have to get the GPS code to work with positionControl() and that has been very difficult.  We can perhaps do this when we see that PR.

Reply:
The arguments are OK. But this function must e called repeatedly to perform the control task. The is suboptimal design WRT control flow. It is beter to make control algorithm 'fire-and-forget' to some extent - provide setpoint and let control algorithm reach it, even without further intervention.


For example this code would fail spectacularly if altitudeControl is called from two places at the same time.

Reply:
Maybe we can look at improving this in a subsequent PR.

Reply:
I also tend to think that moving the target altitude gives better control than trying to control velocity, given the unstable nature of our altitude data. Chasing an intrinsically unstable velocity didn’t feel as good when I tried it out. It was easier when trying to raise the target altitude with the sticks to do exactly that.